web/modules/custom/dpl_opening_hours/src/Plugin/rest/resource/v1/OpeningHoursLegacyResource.php

```diff
@@ -0,0 +1,251 @@
+      OpeningHoursLegacyResponse {
+        return $this->toLegacyResponse($instance);
+      }, $openingHoursInstances);
+
+      $context = (new DefaultSerializationContextFactory())
+        ->createSerializationContext();
+      $context->setSerializeNull(TRUE);
+
+      // Cast the serializer to the custom serializer type, to access the custom
+      // method and satisfy phpstan.
+      $serializer = $this->serializer;
+      /** @var \Drupal\dpl_opening_hours\Plugin\rest\resource\v1\CustomContextSerializer $serializer */
+
+      return (new CacheableResponse($serializer->serializeWithCustomContext($responseData,
+        $this->serializerFormat($request), $context)))
+        ->addCacheableDependency($this->cachableMetadata());
+    }
+    catch (\TypeError $e) {
+      throw new BadRequestHttpException("Invalid input: {$e->getMessage()}",);
+    }
+
+  }
+
+  /**
+   * Map a value object to an OpenAPI response.
+   *
+   * @throws \Exception
+   */
+  public function toLegacyResponse(OpeningHoursInstance $instance) : OpeningHoursLegacyResponse {
+    return (new OpeningHoursLegacyResponse())
+      ->setNid(intval($instance->branch->id()))
+      ->setCategoryTid(intval($instance->categoryTerm->id()))
+      ->setDate(new DateTime($instance->startTime->format('Y-m-d')))
+      ->setStartTime($instance->startTime->format("H:i"))
+      ->setEndTime($instance->endTime->format('H:i'))
+      ->setNotice(NULL);
+  }
+
+  /**
+   * Generate a schema for an opening hours legacy instance.
+   *
+   * This allows for reuse across implementing classes.
+   *
+   * @return mixed[]
+   *   OpenAPI schema for a single opening hours legacy instance.
+   */
+  protected function openingHoursLegacyInstanceSchema(): array {
+    return [
+      "type" => "object",
+      "properties" => [
+        "nid" => [
+          "type" => "integer",
+          "description" => "The node Id of the branch the opening hours instance belongs to.",
+        ],
+        "category_tid" => [
+          "type" => "integer",
+          "description" => "The (t)id of the opening hours category.",
+        ],
+        "date" => [
+          "type" => "string",
+          "format" => "date",
+          "description" => "The date which the opening hours applies to. In ISO 8601 format.",
+        ],
+        "start_time" => [
+          "type" => "string",
+          "example" => "09:00",
+          "description" => "When the opening hours start. In format HH:MM",
+        ],
+        "end_time" => [
+          "type" => "string",
+          "example" => "17:00",
+          "description" => "When the opening hours end. In format HH:MM",
+        ],
+        "notice" => [
+          "type" => "string",
+          "description" => "The id for the branch the instance belongs to",
+        ],
+      ],
+      "required" => [
+        "nid",
+        "category_tid",
+        "date",
+        "start_time",
+        "end_time",
+        "notice",
+      ],
```

Comment:
Please reconsider the definition of the notice part.

1. The description does not seem right. Perhaps a little bit of copy/paste here? :smile:
2. The field is marked as required but we explicitly set it to null - which is not a string as declared by the schema. That does not seem right. I suggest you look into marking the value nullable in the schema.